### ISSUE TITLE
AWS::Config::ConfigRule support tagging

### ISSUE THREAD

**Issue by robpickerill** [aws-cloudformation/aws-cloudformation-coverage-roadmap#117]
1. Title -> AWS::Config::ConfigRule

2. Scope of request -> Cloudformation resource AWS::Config::ConfigRule doesn't appear to support tags: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-config-configrule.html but the API does support tags: https://docs.aws.amazon.com/config/latest/APIReference/API_PutConfigRule.html#config-PutConfigRule-request-Tags

3. Expected behavior -> It should be possible to tag a config rule (as the IAM conditions for PutConfigRule are based on tags).

4. Test case recommendation (optional)

5. Links to existing API doc (optional) -> 
PutConfigRule API - https://docs.aws.amazon.com/config/latest/APIReference/API_PutConfigRule.html#config-PutConfigRule-request-Tags
CFN Config Rule - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-config-configrule.html

6. Category tag (optional) -> Management

7. Any additional context (optional)

**Comment by pgarbe** [comment]
Any news about this one? Having tags helps a lot to protect some organization-wide rules from being modified while having the freedom to add additional rules in the local account.

**Comment by dacacioa** [comment]
Hi team,

Any news? Tags are essential to manage and organize rules.

**Comment by daknhh** [comment]
Any news here? It would be great to have that feature to secure organization-wide rules via SCP without the need to deploy a custom resource to tag the rules

**Comment by kirnberger1980** [comment]
Any news, if this issue is going to be planned on the roadmap? We definitely need this feature.

**Comment by cool-raj** [comment]
Any updates guys... We are blocked and we are building always alternate solutions which are pretty complex and the tagging could just solve the issue.

**Comment by guipaivanz** [comment]
Is there an update on this one? It is been a year since the last comment

**Comment by elduds** [comment]
> Any news here? It would be great to have that feature to secure organization-wide rules via SCP without the need to deploy a custom resource to tag the rules

Exactly. Stuck between a rock and a hard place when org rules (which can be protected in a security account) don't support remediation, but local rules (which run locally in less-trusted accounts) do not support protection from modification via SCP.

**Comment by mattspad-bofa** [comment]
There doesn't seem to be any official stance on this, however, here is my workaround using Python to add a Rule tag:

`import boto3

config_client = boto3.client('config')

# Define the rule name and tags
rule_name = 'required-tags-ou-name'
tags = [
    {
        'Key': 'TestKey',
        'Value': 'TestValue'
    },
    {
        'Key': 'TestKey2',
        'Value': 'TestValue2'
    }
]

try:
    # Get the rule details
    response = config_client.describe_config_rules(
        ConfigRuleNames=[rule_name]
    )

    # Check if the rule exists
    if response['ConfigRules']:
        rule = response['ConfigRules'][0]

        # Add tags to the rule
        config_client.tag_resource(
            ResourceArn=rule['ConfigRuleArn'],
            Tags=tags
        )

        print(f"Tags added successfully to the rule: {rule_name}")
    else:
        print(f"Rule '{rule_name}' not found.")

except Exception as e:
    print(f"Error: {e}")`
